server/data.py

```diff
@@ -268,25 +274,27 @@ def _count_gt_for_write(self, variants, data, filters, chips, data_type):
                 het_i, hom_alt_i, info = self._get_het_hom_index(d, id_index, filters['gtgp'] == 'gt', filters['gpThres'], len(data) == 1)
             else:
                 het_i, hom_alt_i, info = self._get_het_hom_index(d, id_index, True, None, False)
-            # extract gt probs and gts
-            gt_probs = [element.split(':')[2] for element in d ]
             gt = [element.split(':')[0] for element in d ]
-            gt_probs_arr = np.array(gt_probs)
-            gt_probs_het = list(gt_probs_arr[het_i])
-            gt_probs_hom = list(gt_probs_arr[hom_alt_i])
             gt_arr = np.array(gt)
             gt_het = list(gt_arr[het_i])
             gt_hom = list(gt_arr[hom_alt_i])
 
             # subset dataframe for het/hom individuals, copy needed as this will be mutated
-            het = self.info_orig.iloc[het_i].copy()
-            hom_alt = self.info_orig.iloc[hom_alt_i].copy()
+            het = info_orig.iloc[het_i].copy()
+            hom_alt = info_orig.iloc[hom_alt_i].copy()
+
+            # extract gt probs and gts
+            if data_type == 'imputed':
+                gt_probs = [element.split(':')[2] for element in d ]
```

Comment:
general thing in reading data. We don't know in what order and which fields might be present in genotype fields. When reading in variants the fields should be read from FORMAT column. 
VCF specs here for clarification. https://samtools.github.io/hts-specs/VCFv4.2.pdf.

Parsing of these fields should happen in _get_genotype_data and e.g. returning a list of dicts with all those fields or at least dict of available fields and their index.

Reply:
anyway this change is needed as the GP fields just don't exist.